packages/babylon/src/parser/expression.js

```diff
@@ -645,7 +645,10 @@ export default class ExpressionParser extends LValParser {
           !this.state.inClassProperty &&
           !this.options.allowSuperOutsideMethod
         ) {
-          this.raise(this.state.start, "'super' outside of function or class");
+          this.raise(
+            this.state.start,
```

Comment:
just me but I think [[HomeObject]] is going to be confusing to people, few users are going to know what that means.

The issue is originally that they are using `super.` in a function expression rather than an object method

Reply:
What would you suggest?

Reply:
`super is only allowed in object methods and classes`?

Reply:
It make sense. Will update the PR

Reply:
Updated the PR. But we can make a blog about [[HomeObject]] to make sure the peoples under the internals.

Reply:
SM uses `super() is only valid in derived class constructors`. V8 similiar.

Reply:
That message doesn't work with `super.*`. Babylon already throws `super() is only valid inside a class constructor. Make sure the method name is spelled exactly as 'constructor'.`

Reply:
Well. What about `Member access on super must be in a method`  or `Calls to super must be in the "constructor" method of a class expression or class declaration that has a superclass`.

Reply:
@nicolo-ribaudo where do you get this message?

Reply:
https://github.com/babel/babel/blob/9ae23639ad12898e3bf2dff7b1d1a9fbd2be15fe/packages/babylon/src/parser/expression.js#L670

Actually it looks like it is only thrown when `super()` is used in non-constructor methods. It would make sense to defer the `super outside a class` error (https://github.com/babel/babel/blob/9ae23639ad12898e3bf2dff7b1d1a9fbd2be15fe/packages/babylon/src/parser/expression.js#L646-L652) and throw the correct error based on the next token (`.` or `(`).

Reply:
Make sense. Thanks for pointing out.